Issue #546 - Add possibility in switch node

Mathiponds opened the pull request:
It is now possible to have a `switch` node indented as
```java
switch(x)
case 1:
    return y;
case 2:
    return z;
}
```
You have to set the parameter `indentInSwitch` of
`PrettyPrinterConfiguration` to do so. Its default value is `true`

Fixes #546.

Mathiponds committed:
Add possibility in switch node

It is now possible to have a switch node indented as
```java
switch(x)
case 1:
    return y;
case 2:
    return z;
}
```
You have to set the parameter `indentInSwitch` of
`PrettyPrinterConfiguration` to do so. Its default value is `true`

Mathiponds committed:
Merge branch 'master' into issue-546

Mathiponds commented on javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrintVisitor.java:
```diff
@@ -1164,11 +1164,15 @@ private void printSwitchNode(SwitchNode n, Void arg) {
+            if(configuration.isIndentInSwitch()) {
+                printer.unindent();
+            }
         }
```
Or maybe it is better to write it as 
```java 
            if(configuration.isIndentInSwitch()) {
                printer.indent();
                for (final SwitchEntry e : n.getEntries())
                    e.accept(this, arg);
                print.unindent();
            }else{
                for (final SwitchEntry e : n.getEntries()) 
                    e.accept(this, arg);
            }
```

MysterAitch replied:
Personally, I think a cleaner/tidier alternative would be:

```java
printer.indent(configuration.isIndentInSwitch());
```

Or a helper method `indentIf(boolean shouldIndent)` (or similar).

Mathiponds replied:
Good idea I will do that

matozoid replied:
I don't see why the indenting printer should be modified for a single case?

Mathiponds committed:
Change name, add javadoc, add helper method

Following the comments on the PR, I change the name `indentInSwitch` to `indentCaseInSwitch`
I added a enxplanation on what this variable changes
I added two private methods `indentIf` and `unindentIf` that take a
boolean and do the indentation if the argument is true.

matozoid committed:
Merge branch 'master' into issue-546

matozoid committed:
Merge branch 'master' into issue-546
